File: apps/frontend/mobile/src/hooks/club/clubView.ts
import type { MobileClubDetail } from '@/src/api/generated/mobile-gateway/models';
import type { Club } from '@/src/types/Club';

/** Projects the privacy-filtered canonical club response into the profile view. */
export function toClubView(response: MobileClubDetail): Club {
  return {
    id: response.id,
    rawName: response.rawName,
    name: response.name,
    city: response.city,
    email: response.email,
    phoneNumber: null,
    website: response.website,
    address: response.address,
    logoUrl: response.logoUrl,
    longitude: response.longitude,
    latitude: response.latitude,
  };
}

File: apps/backend/clubs-service/src/main/java/com/blockout/clubs/club/application/views/ClubView.java
package com.blockout.clubs.club.application.views;

import java.time.LocalDateTime;

/**
 * Complete application view of a club.
 */
public record ClubView(
    String id,
    String rawName,
    String name,
    String address,
    String city,
    String postalCode,
    String email,
    String phoneNumber,
    String website,
    String logoUrl,
    boolean active,
    Double latitude,
    Double longitude,
    LocalDateTime createdAt,
    LocalDateTime lastUpdate) {
}
